modules/preview-react/mutliselect/MultiSelectInput.tsx

```diff
@@ -0,0 +1,306 @@
+  createElemPropsHook(useMultiSelectModel)((model, ref) => {
+    return {
+      onKeyDown(event: React.KeyboardEvent) {
+        console.log('onKeyDown', event.key);
+        // Space bar key is hit and menu is open
+        if (event.key === ' ' && model.state.visibility === 'visible') {
+          const id = model.state.cursorId;
+          // There are 2 modes. Search mode and navigation mode. Searching clears the cursor. Using
+          // the down arrow enables the cursor and thus navigation mode.
+          // - Navigation mode: toggle cursor item
+          // - Search mode: enter a space in the search
+          if (id) {
+            model.events.select({id});
+            event.preventDefault();
+          }
+        }
+
+        // console.log('key', event.key);
+        if (
+          (event.key === 'ArrowDown' || event.key === 'ArrowUp') &&
+          model.state.visibility === 'hidden'
+        ) {
+          // prevent navigating vertically on the input
+          model.events.show(event);
+          event.preventDefault();
+        }
+      },
+      'aria-haspopup': 'listbox' as const,
+      onChange: e => console.log('user-input', e.currentTarget.value),
+      // onClick(e: React.MouseEvent) {
+      //   console.log('click', model.state.visibility);
+      //   if (model.state.visibility === 'hidden') {
+      //     model.events.show(e);
+      //   } else if (model.state.visibility === 'visible') {
+      //     model.events.hide(e);
+      //   }
+      // },
+    };
+  }),
+  useComboboxInputConstrained,
+  useComboboxInput
+);
+
+const removeItem = <T extends unknown>(id: string, model: ReturnType<typeof useListModel>) => {
+  const index = model.state.items.findIndex(item => item.id === model.state.cursorId);
+  const nextIndex = index === model.state.items.length - 1 ? index - 1 : index + 1;
+  const nextId = model.state.items[nextIndex].id;
+  console.log('nextId', id, nextId);
+  if (model.state.cursorId === id) {
+    // We're removing the currently focused item. Focus next item
+    model.events.goTo({id: nextId});
+
+    // // wait for stabilization of state
+    // requestAnimationFrame(() => {
+    //   document.querySelector<HTMLElement>(`#${model.state.id}-${nextId}`)?.focus();
+    // });
+  }
+};
+
+const useMultiSelectedItem = composeHooks(
+  createElemPropsHook(useListModel)((model, ref, elemProps) => {
+    return {
+      onKeyDown(event: React.KeyboardEvent<HTMLElement>) {
+        const id = event.currentTarget.dataset.id || '';
+        if (event.key === 'Backspace' || event.key === 'Delete') {
+          model.events.select({id});
+          removeItem(id, model);
+        }
+      },
+      onClick(event: React.MouseEvent<HTMLElement>) {
+        const id = event.currentTarget.dataset.id || '';
+        model.events.select({id});
+      },
+    };
+  }),
+  useListItemRovingFocus,
+  useListItemRegister
+);
+
+const MultiSelectedItem = createSubcomponent('span')({
```

Comment:
```suggestion
export const MultiSelectedItem = createSubcomponent('span')({
```

Reply:
No. We do not export subcomponents. They need to be attached to a container instead.